server/user/user.controller.js

```diff
@@ -44,9 +44,32 @@ function login({ body }, res) {
     });
 }
 
+function updateProfile({ user, body: { userCredentials } }, res, next) {
+  const { email, firstName, lastName } = userCredentials;
```

Comment:
`credentials` means `username` plus `password` so please rename it accordingly. Also don't use nested destructuring which is totally unneeded here btw. Just put `email`, `firstName` and `lastName` inside `body` directly.